wallet/src/service/Offers.ts

```diff
@@ -117,15 +123,36 @@ export const getOfferService = (
       Object.fromEntries(
         await Promise.all(
           Object.entries(proposalTemplate).map(async ([kw, entry]) => {
-            if (entry.amount && !(entry.pursePetname && entry.value)) {
-              const unserializedAmount = await E(boardIdMarshaller).unserialize(
-                entry.amount,
-              );
-              entry.pursePetname = brandToPurse.get(unserializedAmount.brand)
-                ?.pursePetname;
-              entry.value = String(unserializedAmount.value);
+            assert(entry.amount);
+            const unserializedAmount = await E(boardIdMarshaller).unserialize(
+              entry.amount,
+            );
+            const purse = brandToPurse.get(unserializedAmount.brand);
+
+            if (purse) {
+              const { pursePetname: brandPetname, displayInfo } = purse;
+              return [
+                kw,
+                { amount: unserializedAmount, brandPetname, displayInfo },
+              ];
             }
-            return [kw, entry];
+
+            const [brands, boardAux] = await Promise.all([
+              brandsP,
+              watcher.queryBoardAux<{
+                displayInfo: DisplayInfo;
+              }>([unserializedAmount.brand])[0],
+            ]);
+            const brandPetname = brands
+              .find(([_, brand]) => brand === unserializedAmount.brand)
+              ?.at(0);
```

Comment:
is `at` because `[]` doesn't work with `?.` ?

please find or make a general utility for taking the one element of an array, validating that it is defined and has length exactly one.

Reply:
I like the idea of such a utility. There are many places where we should be using this to be more robust against surprises, especially the numerous places we assume without checking that an invitations payment amount carries only one invitations details.

In other systems, I have called such a function `theOne`. May seem silly but reads well.